runtime/v2/README.md

```diff
@@ -1,29 +1,174 @@
 # Runtime v2
 
 Runtime v2 introduces a first class shim API for runtime authors to integrate with containerd.
+
+containerd, the daemon, does not directly launch containers. Instead, it acts as a higher-level manager
+or hub for coordinating the activities of containers and content, that lower-level
+programs, called "runtime shims" and "runtimes", actually implement to start, stop and manage individual containers
+and container groups called pods.
+For example, containerd will retrieve container image config and its content as layers, lay it out on disk, set up
+the container's rootfs and config, and then create a shim instance that uses the runtime engine,
+for example `runc`, to actually create/start/stop the container.
+
+This document describes the major components of the v2 runtime shim integration model, how the components interact
+with containerd and the v2 shim, and how to use and integrate different v2 shims and runtime engines.
+
+To simplify the interaction, runtime v2 introduced a first class v2 shim API for runtime authors to integrate with containerd,
+replacing the v1 shim api.
 The shim API is minimal and scoped to the execution lifecycle of a container.
 
-## Binary Naming
+This document is split into the following sections:
+
+* [architecture](#architecture) - the major components, their purposes and relationships
+* [usage](#usage) - how to invoke specific runtimes, and how to configure them
+* [authoring](#shim-authoring) - how to author a v2 shim
+
+## Architecture
+
+### containerd-runtime communication
+
+containerd expects a runtime to implement several container control features, such as create, start and stop.
+The most common runtime used is [runc](https://github.com/opencontainers/runc).
```

Comment:
I am not sure what you are suggesting. That the split between runtime engine and runtime shim is not inherent to containerd, which only cares about, "I have a runtime that speaks by API, I talk to it, I don't care how it goes about doing it (split shim and engine, not split, don't care)"? I get that, and I guess (defer to you, of course) that is technically correct.

In practice, it is split, including components shipped by containerd (the runc shim). Again, in the first review, @mikebrow commented to make the split even more clear. So happy to go with what consensus is, as long as we can get that.

Reply:
containerd expects the shim to implement [this interface](https://github.com/containerd/containerd/blob/b30e0163ac36c1a193604e5eca031053d62019c5/api/runtime/task/v2/shim.proto#L33-L51).

That _can_ be implemented by a shim which invokes an OCI runtime (such as runc).  It's just that containerd doesn't "expect" it.  So my suggestion is to rephrase this to avoid the implication that containerd expects that split or the specific OCI runtime spec implementation.

Reply:
Let's let @mikebrow weigh in, as he did an extensive and helpful review (not to mention, describing all of this to me on Slack, so I could understand enough to solve my specific issue, which led to my trying to be helpful to future people by documenting it all).

I think the best approach may be my suggestion above: make it clear here what the border between containerd and "runtime" is, reference in a paragraph that many split shim and engine, and then reference details in a different doc.

Reply:
Windows default path right now merges the shim and runc compatible runtime engine into one binary.. and on linux our default v2 path is the way it's described.  Sam is right language should be coached to indicate flexibility in this regard.  E.g. at the top where describing such detail mention what Sam suggests then you can say something like for the default linux shimv2 yada yada the components are split into..

Reply:
I'd still like to see this addressed.

Reply:
I am pretty sure we did. Take a look at the latest version.

Reply:
The language here still talks about containerd expecting runc to implement container operations, which isn't correct.

Reply:
I think we should just remove `runc` from there, as it is confusing. Coming in a few mins

Reply:
I tried some restructuring there. Have a look.